test(frontend): Cover practically all the Storybook tests with snapshots

Twixes opened the pull request:
## Changes

Wanted to enable snapshots just for the Saved Insights stories, since I'm gonna work on that. But since #13947 made snapshots more reliable, why not enable this for even more stories.

Twixes committed:
test(frontend): Moar story snapshots

Twixes committed:
Merge branch 'master' into saved-insights-vrt

Twixes committed:
Convert .storybook/main.js to TypeScript

Twixes committed:
Always use `targets: 'defaults'` for frontend Babel

Twixes committed:
Enable snapshots for a bunch more stories

Twixes committed:
Fix comment

Twixes committed:
Use `RETRY_TIMES` properly

Twixes committed:
Fix typing

Twixes committed:
Fix User Paths rendering

Twixes committed:
Hide "misconfigured site URL" warning in Storybook

Twixes committed:
Stabilize hour in `SubscriptionsModal` stories

Twixes committed:
Update Navigation snapshots

Twixes commented on frontend/src/scenes/paths/Paths.tsx:
```diff
@@ -43,7 +43,9 @@ export function PathsComponent({ nodeCard }: PathsComponentProps): JSX.Element {
         setNodeCards([])
 
-        const elements = document?.getElementById(id)?.querySelectorAll(`.Paths svg`)
+        // Remove the existing SVG canvas(es). The .Paths__canvas selector is crucial, as we have to be sure
+        // we're only removing the Paths viz and not, for example, button icons.
```
Already discovered one weird bug thanks to these tests: turns out resizing the window or toggling the sidebar resulted in the User Paths node card kebab button (dropdown) being removed 🤔

thmsobrmlr replied:
Haha, good catch!

Twixes committed:
Update scenes-app-web-performance--web-performance.png

thmsobrmlr commented on frontend/src/lib/components/HedgehogBuddy/HedgehogBuddy.stories.tsx:
```diff
@@ -5,7 +5,7 @@ export default {
     component: HedgehogBuddy,
     parameters: {
-        chromatic: { disableSnapshot: true },
+        testOptions: { skip: true }, // Hedgehogs aren't particularly snapshotable
```
That's very offensive to hedgehogs

thmsobrmlr commented on frontend/src/scenes/PreflightCheck/preflightLogic.tsx:
```diff
@@ -214,7 +214,10 @@ export const preflightLogic = kea<preflightLogicType>([
             (preflight): boolean => {
-                return Boolean(preflight && (!preflight.site_url || preflight.site_url != window.location.origin))
+                if (window.location.port === '6006') {
+                    return false // Disable this warning in Storybook
```
Not sure which warning that is, but this won't work for https://storybook.posthog.net/

Twixes replied:
Good catch, updated this "fix" to work in that case too, and explained the reason

thmsobrmlr reviewed (approved):
Loving all these little improvements! Feels great to have all the stories tested, the storybook config in typescript, a paths bugfix, etc.

We should remove the duplicated tests (insights) from e2e-vrt (I can do that after this is in). scenes-app-insights--user-paths.png has been flaky for snapshots in the past, so let's see how this goes.

Twixes committed:
Clean up `siteUrlMisconfigured` Storybook fix

Twixes committed:
Optimize `Dockerfile.playwright`

I kept running out of disk space because any change to frontend code resulted in a whole new `Dockerfile.playwright` image being generated. We don't actually need to copy the frontend/ directory - let's just mount it instead.

Twixes committed:
Reduce opportunities for insight snapshot flakiness

Twixes committed:
Merge branch 'master' into saved-insights-vrt

Twixes committed:
Merge branch 'master' into saved-insights-vrt
